python/beeai_framework/backend/constants.py

```diff
@@ -37,4 +37,5 @@ class ProviderModelDef(BaseModel):
     "Ollama": ProviderDef(name="Ollama", module="ollama", aliases=[]),
     "OpenAI": ProviderDef(name="OpenAI", module="openai", aliases=["openai"]),
     "watsonx": ProviderDef(name="Watsonx", module="watsonx", aliases=["watsonx", "ibm"]),
+    "xAI": ProviderDef(name="XAI", module="xai", aliases=["xai", "grok"]),
```

Comment:
Isn't the name `xAI`?

Reply:
It is. However there's a challenge with the code that maps the provider name to model (see [this comment](https://github.com/i-am-bee/beeai-framework/pull/446#issuecomment-2692935104) (also need to be compliant to PEP8 for upper class first letter in class name)